SD-K9/src/app/components/outdoor-map/outdoor-map.component.ts

```diff
@@ -18,26 +22,35 @@ export class OutdoorMapComponent implements OnInit, OnDestroy, Map {
   @Input() data: any;
   @ViewChild('map', {static: true}) mapElement: ElementRef;
   map: any;
+  //cannot set type to google.maps.marker because google maps is not loaded yet
+  buildingMarkers: any[] = [];
   userMarker: any;
+  buildingInfoCardIsShown: boolean = false;
+  currentBuilding: string;
+  currentCampus: string;
+  currentBuildingInfo: string;
+  campusConfig: any = campusData["default"];
   mapInitialised: boolean = false;
   currentPos: GoogleCoordinate;
   apiKey: string = "AIzaSyA_u2fkanThpKMP4XxqLVfT9uK0puEfRns";
```

Comment:
This is dangerous since the repo is public and anyone can see the key.
I suggest using angular's environment variables to store this and ignoring the file in gitignore.

**inside environments.ts:**
~~~
export const environment = {
  production: false,
  apiKey: 'the api key goes here'
};
~~~
**inside outdoor-map-component.ts:**
~~~
import { environment } from '../../../environments/environment';
apiKey: string = environment.apiKey;
~~~